neuralop/models/base_model.py

```diff
@@ -76,7 +76,38 @@ def __new__(cls, *args, **kwargs):
         instance._init_kwargs = kwargs
 
         return instance
-    
+
+    def state_dict(self, **kwargs):
+        """
+        state_dict subclasses nn.Module.state_dict() and adds a metadata field
+        to track the model version and ensure only compatible saves are loaded.
+        """
+        print("In state dict")
+        state_dict = super().state_dict(**kwargs)
+        if state_dict.get('_metadata') == None:
+            state_dict['_metadata'] = self._init_kwargs
+        else:
+            warnings.warn("Attempting to update metadata for a module with metadata already in self.state_dict()")
+        return state_dict
+
+    def load_state_dict(self, state_dict, **kwargs):
+        """
+        state_dict subclasses nn.Module.load_state_dict() and adds a metadata field
+        to track the model version and ensure only compatible saves are loaded.
+        """
+        metadata = state_dict.pop('_metadata', None)
+        if metadata is not None:
+
+            saved_version = metadata.get('_version', None)
+            if saved_version is None:
+                warnings.warn(f"Saved instance of {self.__class__} has no stored version attribute.")
+            self_version = self._init_kwargs['_version']
+            if saved_version != self_version:
```

Comment:
```suggestion
            if saved_version != self._version:
```
and let's omit the line defining self_version above